dotnet/src/IntegrationTests/Connectors/Memory/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreFixture.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.SemanticKernel.Data;
+using MongoDB.Driver;
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Connectors.AzureCosmosDBMongoDB;
+
+public class AzureCosmosDBMongoDBVectorStoreFixture : IAsyncLifetime
+{
+    private readonly List<string> _testCollections = ["sk-test-hotels", "sk-test-contacts", "sk-test-addresses"];
+
+    /// <summary>Main test collection for tests.</summary>
+    public string TestCollection => this._testCollections[0];
+
+    /// <summary><see cref="IMongoDatabase"/> that can be used to manage the collections in Azure CosmosDB MongoDB.</summary>
+    public IMongoDatabase MongoDatabase { get; }
+
+    /// <summary>Gets the manually created vector store record definition for Azure CosmosDB MongoDB test model.</summary>
+    public VectorStoreRecordDefinition HotelVectorStoreRecordDefinition { get; private set; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureCosmosDBMongoDBVectorStoreFixture"/> class.
+    /// </summary>
+    public AzureCosmosDBMongoDBVectorStoreFixture()
+    {
+        var configuration = new ConfigurationBuilder()
+            .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
+            .AddJsonFile(
+                path: "testsettings.development.json",
+                optional: false,
+                reloadOnChange: true
+            )
+            .AddEnvironmentVariables()
+            .Build();
+
+        var connectionString = GetConnectionString(configuration);
+        var client = new MongoClient(connectionString);
+
+        this.MongoDatabase = client.GetDatabase("test");
+
+        this.HotelVectorStoreRecordDefinition = new()
+        {
+            Properties =
+            [
+                new VectorStoreRecordKeyProperty("HotelId"),
+                new VectorStoreRecordDataProperty("HotelName") { PropertyType = typeof(string) },
+                new VectorStoreRecordDataProperty("HotelCode") { PropertyType = typeof(int) },
+                new VectorStoreRecordDataProperty("ParkingIncluded") { PropertyType = typeof(bool), StoragePropertyName = "parking_is_included" },
+                new VectorStoreRecordDataProperty("HotelRating") { PropertyType = typeof(float) },
+                new VectorStoreRecordDataProperty("Tags"),
+                new VectorStoreRecordDataProperty("Description"),
+                new VectorStoreRecordVectorProperty("DescriptionEmbedding") { Dimensions = 4, IndexKind = "vector-ivf", DistanceFunction = "COS" }
```

Comment:
We should be able to use the `DistanceFunction.CosineSimilarity` or `.DistanceFunctionCosineDistance` values here.

Reply:
Same reply as here: https://github.com/microsoft/semantic-kernel/pull/7557#discussion_r1698877157